File: algorithm/496. Next Greater Element I/496.py
class Solution(object):
    def nextGreaterElement(self, findNums, nums):
        """
        :type findNums: List[int]
        :type nums: List[int]
        :rtype: List[int]
        """
        m={}
        s=[]
        ans=[]
        for i in nums:
            while len(s) and s[-1]<i:
                m[s.pop()]=i
            s.append(i)
        for i in findNums:
            ans.append(m.get(i, -1))
        return ans

File: algorithm/496. Next Greater Element I/496.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& findNums, vector<int>& nums) {
        stack<int> s;
        map<int,int> m;
        for(int i : nums)
        {
            while(!s.empty() && s.top()<i)
            {
                m[s.top()]=i;
                s.pop();
            }
            s.push(i);
        }
        vector<int> result;
        for(int i:findNums)
        {
            if(m.count(i)!=0)
                result.push_back(m[i]);
            else
                result.push_back(-1);
        }
        return result;
    }
};
